Mathlib/Analysis/SpecialFunctions/Log/ENNRealLog.lean

```diff
@@ -0,0 +1,228 @@
+  · simp
+  · rw [log_top]
+    rcases ENNReal.trichotomy y with (rfl | rfl | y_real)
+    · rw [mul_zero, log_zero, EReal.add_bot]
+    · simp
+    · rw [log_pos_real' y_real, ENNReal.top_mul', EReal.top_add_coe, log_eq_top_iff]
+      simp only [ite_eq_right_iff, zero_ne_top, imp_false]
+      exact Ne.symm (ne_of_lt (ENNReal.toReal_pos_iff.1 y_real).1)
+  · rw [log_pos_real' x_real]
+    rcases ENNReal.trichotomy y with (rfl | rfl | y_real)
+    · simp
+    · simp [Ne.symm (ne_of_lt (ENNReal.toReal_pos_iff.1 x_real).1)]
+    · have xy_real := Real.mul_pos x_real y_real
+      rw [← ENNReal.toReal_mul] at xy_real
+      rw_mod_cast [log_pos_real' xy_real, log_pos_real' y_real, ENNReal.toReal_mul]
+      exact Real.log_mul (Ne.symm (ne_of_lt x_real)) (Ne.symm (ne_of_lt y_real))
+
+theorem log_pow {x : ℝ≥0∞} {n : ℕ} : log (x ^ n) = (n : ℝ≥0∞) * log x := by
+  cases' Nat.eq_zero_or_pos n with n_zero n_pos
+  · simp [n_zero, pow_zero x]
+  rcases ENNReal.trichotomy x with (rfl | rfl | x_real)
+  · rw [zero_pow (Ne.symm (ne_of_lt n_pos)), log_zero, EReal.mul_bot_of_pos]; norm_cast
+  · rw [ENNReal.top_pow n_pos, log_top, EReal.mul_top_of_pos]; norm_cast
+  · replace x_real := ENNReal.toReal_pos_iff.1 x_real
+    have x_ne_zero := Ne.symm (LT.lt.ne x_real.1)
+    have x_ne_top := LT.lt.ne x_real.2
+    simp only [log, pow_eq_zero_iff', x_ne_zero, ne_eq, false_and, ↓reduceIte, pow_eq_top_iff,
+      x_ne_top, toReal_pow, Real.log_pow, EReal.coe_mul]
+    rfl
+
+theorem log_rpow {x : ℝ≥0∞} {y : ℝ} : log (x ^ y) = y * log x := by
+  rcases lt_trichotomy y 0 with (y_neg | rfl | y_pos)
+  · rcases ENNReal.trichotomy x with (rfl | rfl | x_real)
+    · simp only [ENNReal.zero_rpow_def y, not_lt_of_lt y_neg, ne_of_lt y_neg, log_top, log_zero]
+      exact Eq.symm (EReal.coe_mul_bot_of_neg y_neg)
+    · rw [ENNReal.top_rpow_of_neg y_neg, log_zero, log_top]
+      exact Eq.symm (EReal.coe_mul_top_of_neg y_neg)
+    · have x_ne_zero := Ne.symm (LT.lt.ne (ENNReal.toReal_pos_iff.1 x_real).1)
+      have x_ne_top := LT.lt.ne (ENNReal.toReal_pos_iff.1 x_real).2
+      simp only [log, rpow_eq_zero_iff, x_ne_zero, false_and, x_ne_top, or_self, ↓reduceIte,
+        rpow_eq_top_iff]
+      norm_cast
+      exact ENNReal.toReal_rpow x y ▸ Real.log_rpow x_real y
+  · simp
+  · rcases ENNReal.trichotomy x with (rfl | rfl | x_real)
+    · rw [ENNReal.zero_rpow_of_pos y_pos, log_zero, EReal.mul_bot_of_pos]; norm_cast
+    · rw [ENNReal.top_rpow_of_pos y_pos, log_top, EReal.mul_top_of_pos]; norm_cast
+    · have x_ne_zero := Ne.symm (LT.lt.ne (ENNReal.toReal_pos_iff.1 x_real).1)
+      have x_ne_top := LT.lt.ne (ENNReal.toReal_pos_iff.1 x_real).2
+      simp only [log, rpow_eq_zero_iff, x_ne_zero, false_and, x_ne_top, or_self, ↓reduceIte,
+        rpow_eq_top_iff]
+      norm_cast
+      exact ENNReal.toReal_rpow x y ▸ Real.log_rpow x_real y
+
+end Morphism
+
+/-! ### Topological properties -/
+
+section Continuity
+
+/-- `log` as an homeomorphism. --/
```

Comment:
```suggestion
/-- `log` as a homeomorphism. -/
```
The h at the beginning of "homeomorphism" is pronounced, so it starts with a consonant.